addons/web/static/src/js/services/data_manager.js

```diff
@@ -414,6 +418,15 @@ return core.Class.extend({
             if (node.tag === 'field') {
                 fieldsInfo[node.attrs.name] = self._processField(viewType,
                     fields[node.attrs.name], node.attrs ? _.clone(node.attrs) : {});
+
+                if (fieldsInfo[node.attrs.name].fieldDependencies) {
+                    var deps = fieldsInfo[node.attrs.name].fieldDependencies;
+                    for (var dependency_name in deps) {
+                        if (!(dependency_name in fieldsInfo)) {
+                            fieldsInfo[dependency_name] = {'name': dependency_name, 'dependency_of': node.attrs.name};
```

Comment:
I think that the dependency_of is not useful, and may be wrong (if you have more than one dependant fields).

Also, you should add the type of the field.  It probably works in a form view, because the basic model knows all the fields from the current model, but be aware that the web client does not know the description of all fields for the sub models (such as a one2many)